fw/procfs.c

```diff
@@ -359,7 +402,57 @@ tfw_procfs_stop(void)
 	tfw_procfs_cleanup();
 }
 
+static int
+tfw_cfgop_health_stat(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	int cpu;
+
+	TFW_CFG_CHECK_VAL_N(>, 0, cs, ce);
+	TFW_CFG_CHECK_NO_ATTRS(cs, ce);
+
+	health_stat_codes = kzalloc(tfw_hm_stats_size(ce->val_n),
+				    GFP_KERNEL);
+	if (!health_stat_codes)
+		return -ENOMEM;
+	if (tfw_hm_stats_init_from_cfg_entry(health_stat_codes, ce)) {
+		kfree(health_stat_codes);
+		health_stat_codes = NULL;
+		return -EINVAL;
+	}
+
+	for_each_online_cpu(cpu) {
```

Comment:
Not sure this is the correct approach. What is one of the initially online CPUs goes down, and another one comes up? What if an additional CPU is inserted (hotplug)?

The same goes for destruction of data allocated here.

Reply:
Good point! I try to increment count of online CPU in qemu and catch a lot of problems with memory allocation/deallocation in all places where we use  `for_each_online_cpu`. (In sock.c also). @krizhanovsky we should fix it here or in other task?

Reply:
Fixed